From 29dc958dfe7c5d7884ddeeff130656babe12bd5a Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 23 Aug 2022 13:30:50 +0300
Subject: [PATCH] [SW-99461] gaudi3: update validation pass jump logic

As the start address is a part of the persistant memory variable,
it would be retained.

Hence it is not required to re-read the image descriptor in case
validation passed to read the start address.

Change-Id: Ief6aa049b4708fc0479fcd648854850f37cb5d42
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 404fbcc..d671bf6 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -887,27 +887,12 @@
 	uint64_t *start_addr =
 			&arcboot_rom_global.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].start_addr;
 	int ret;
-	uint32_t idx;
-	struct boot_desc desc;
 
 	if (!arcboot_rom_boot_desc_init_done(HL_FW_BOOT_PTR_SPI)) {
 		goto skip_bringup;
 	}
 
 	if (arcboot_rom_global.validation_pass) {
-		idx = boot_desc_get_img_desc(HL_FW_BOOT_PTR_SPI,
-				HL_FW_IMG_PPBOOT, &desc, FLOPS_UPD_NOOP);
-
-		/* Toggle the index in case primary index failed */
-		if (idx == HL_FW_IMG_NOT_FOUND)
-			idx = boot_desc_get_img_desc(HL_FW_BOOT_PTR_SPI,
-				HL_FW_IMG_PPBOOT, &desc, FLOPS_UPD_PENDING);
-
-		/* Both index failed. Go to recovery */
-		if ((idx == HL_FW_IMG_NOT_FOUND) || (!desc.start_addr))
-			goto skip_bringup;
-
-		*start_addr = desc.start_addr;
 		goto skip_rom_validation;
 	}
 
